web/src/nav/GlobalNavbar.tsx

```diff
@@ -82,9 +83,14 @@ export class GlobalNavbar extends React.PureComponent<Props, State> {
         super(props)
 
         // Reads initial state from the props (i.e. URL parameters).
-        const navbarQuery = parseSearchURLQuery(props.location.search || '', this.props.interactiveSearchMode, true)
-        if (navbarQuery) {
-            props.onNavbarQueryChange({ query: navbarQuery, cursorPosition: navbarQuery.length })
+        const query = parseSearchURLQuery(props.location.search || '')
+        if (query) {
+            if (props.interactiveSearchMode) {
+                const { navbarQuery } = convertPlainTextToInteractiveQuery(query)
+                props.onNavbarQueryChange({ query: navbarQuery, cursorPosition: navbarQuery.length })
```

Comment:
Is this not needed here?

```suggestion
                props.onNavbarQueryChange({ query: navbarQuery, cursorPosition: navbarQuery.length })
                props.onFiltersInQueryChange(filtersInQuery)
```

Reply:
No, we want the `InteractiveModeInput` to call `onFltersInQueryChange`, since that component is used on the homepage too, and needs to update filtersInQuery when it's not in the navbar. 

I removed the `onFiltersInQueryChange` call in `componentDidUpdate` in this component, since that is handled in InteractiveModeInput as well.